src/commands.rs

```diff
@@ -3867,61 +3868,75 @@ fn cmd_git_push(
 
     let mut tx;
     let mut branch_updates = vec![];
-    if let Some(branch_name) = &args.branch {
-        if let Some(update) = branch_updates_for_push(
-            workspace_command.repo().as_repo_ref(),
-            &args.remote,
-            branch_name,
-        )? {
-            branch_updates.push((branch_name.clone(), update));
-        } else {
-            writeln!(
-                ui,
-                "Branch {}@{} already matches {}",
-                branch_name, &args.remote, branch_name
-            )?;
+    if !args.branch.is_empty() {
+        for branch_name in &args.branch {
+            if let Some(update) = branch_updates_for_push(
+                workspace_command.repo().as_repo_ref(),
+                &args.remote,
+                branch_name,
+            )? {
+                branch_updates.push((branch_name.clone(), update));
+            } else {
+                writeln!(
+                    ui,
+                    "Branch {}@{} already matches {}",
+                    branch_name, &args.remote, branch_name
+                )?;
+            }
         }
         tx = workspace_command.start_transaction(&format!(
-            "push branch {branch_name} to git remote {}",
+            "push {} to git remote {}",
+            make_branch_term(&args.branch),
             &args.remote
         ));
-    } else if let Some(change_str) = &args.change {
-        let commit = workspace_command.resolve_single_rev(change_str)?;
-        let branch_name = format!(
-            "{}{}",
-            ui.settings().push_branch_prefix(),
-            commit.change_id().hex()
-        );
-        if workspace_command
-            .repo()
-            .view()
-            .get_local_branch(&branch_name)
-            .is_none()
-        {
-            writeln!(
-                ui,
-                "Creating branch {} for revision {}",
-                branch_name,
-                change_str.deref()
-            )?;
+    } else if !args.change.is_empty() {
+        let mut commits = vec![];
+        for change_str in &args.change {
+            commits.push(workspace_command.resolve_single_rev(change_str)?);
         }
+        let commits = commits;
```

Comment:
Will the following work as a way of avoiding the mutable vector that the gets shadowed?
```
let commits: Vec<_> = args.change.iter().map(|change_str| {
  workspace_command.resolve_single_rev(change_str)
}).try_collect()?;
```

Reply:
This works great, thanks! I'm actually a little confused how it can work since https://github.com/rust-lang/rust/issues/94047 seems to say the feature is unstable, but it seems this comes from "itertools".